File: app/src/main/java/social/entourage/android/api/model/BaseOrganization.java
package social.entourage.android.api.model;

import android.content.Context;

import androidx.annotation.StringRes;

import java.util.Comparator;

/**
 * Created by mihaiionescu on 01/02/2017.
 */

public abstract class BaseOrganization {

    public static final int TYPE_ORGANIZATION = 0;
    public static final int TYPE_PARTNER = 1;

    public abstract int getType();

    public abstract String getTypeAsString(Context contet);

    public abstract String getName();

    public abstract String getSmallLogoUrl();

    public abstract String getLargeLogoUrl();

    public static class CustomComparator implements Comparator<BaseOrganization> {

        @Override
        public int compare(final BaseOrganization organization1, final BaseOrganization organization2) {
            String name1 = organization1.getName();
            if (name1 == null) return -1;
            String name2 = organization2.getName();
            if (name2 == null) return 1;
            return name1.compareTo(name2);
        }
    }

}

File: app/src/main/java/social/entourage/android/api/model/BaseOrganization.kt
package social.entourage.android.api.model

import android.content.Context

/**
 * Created by mihaiionescu on 01/02/2017.
 */
abstract class BaseOrganization {
    abstract fun getTypeAsString(context: Context): String
    abstract val name: String?
    abstract val largeLogoUrl: String?

    /*class CustomComparator : Comparator<BaseOrganization> {
        override fun compare(organization1: BaseOrganization, organization2: BaseOrganization): Int {
            val name1 = organization1.name ?: return -1
            val name2 = organization2.name ?: return 1
            return name1.compareTo(name2)
        }
    }*/

    companion object {
        const val TYPE_ORGANIZATION = 0
        const val TYPE_PARTNER = 1
    }
}